File: cs/platform.py
"""OS detection and the few platform-specific facts cs needs."""
from __future__ import annotations

import os
import platform as _platform
import sys
from pathlib import Path


def is_macos() -> bool:
    return sys.platform == "darwin"


def is_linux() -> bool:
    return sys.platform.startswith("linux")


def is_wsl() -> bool:
    if not is_linux():
        return False
    try:
        return "microsoft" in Path("/proc/version").read_text().lower()
    except OSError:
        return False


def refuse_unsupported() -> None:
    """Windows-native shells are out of scope; point at WSL."""
    if os.environ.get("MSYSTEM") or sys.platform in ("win32", "cygwin", "msys"):
        raise SystemExit(
            "cs: Windows-native shells are unsupported. Run inside WSL2 "
            "(`wsl --install -d Ubuntu-24.04`). See docs/WINDOWS.md."
        )


def shell_rc() -> Path:
    from .paths import home
    if is_macos():
        return home() / ".zshrc"
    return home() / ".bashrc"


def describe() -> str:
    if is_wsl():
        return "wsl2"
    if is_macos():
        return "macos"
    return _platform.system().lower()

File: src/platform.ts
import { readFileSync } from "node:fs";
import { join } from "node:path";
import { home } from "./paths.js";

export const isMac = () => process.platform === "darwin";
export const isLinux = () => process.platform === "linux";
export function isWSL(): boolean {
  if (!isLinux()) return false;
  try { return readFileSync("/proc/version", "utf8").toLowerCase().includes("microsoft"); } catch { return false; }
}
export function refuseUnsupported() {
  if (process.platform === "win32" || process.env.MSYSTEM) {
    console.error("cs: Windows-native shells are unsupported. Run inside WSL2 (`wsl --install -d Ubuntu-24.04`). See docs/WINDOWS.md.");
    process.exit(1);
  }
}
export const shellRc = () => join(home(), isMac() ? ".zshrc" : ".bashrc");
export const describe = () => (isWSL() ? "wsl2" : isMac() ? "macos" : process.platform);
